tests/test_h2_client.c

```diff
@@ -564,6 +564,7 @@ TEST_CASE(h2_client_stream_with_h1_request_message) {
         DEFINE_HEADER("Accept", "*/*"),
         DEFINE_HEADER("Host", "example.com"),
         DEFINE_HEADER("Content-Length", "5"),
+        DEFINE_HEADER("Transfer-Encoding", "chunked"), /* Connection-specific header should be skiped */
```

Comment:
trivial: use a different connection-specific header.
It's super illegal in HTTP/1 to have both content-length and transfer-encoding